### ISSUE TITLE
[ISSUE] fmDNS truncates slave cache files during build config

### ISSUE THREAD

**Issue by millallo** [WillyXJ/facileManager#574]
fM Version : 4.3.0
fmDNS Version : 5.2.1

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

**(BUG | ISSUE) Expected Behavior:**
Bind cache files on slave servers should not be truncated when building a new configuration.

**(BUG | ISSUE) Actual Behavior:**
I have a hidden master and slave dns authoritative for some zones. If I change some options and build the config all file inside of /var/cache/bind/slave folder are emptied. Bind continues to run on slave without any issue but if I restart it, it renames all truncated files like db-* and recreates all cache files asking them to master server. But what if the master is down? I mean, is it correct to empty the cache while bind is running? I think it's correct to create an empty cache only if cache files don't exist.

**(BUG | ISSUE) Steps to reproduce:**
Build configuration to a running slave.

** Comment **
This is the temporary patch i use:
```
--- a/server/fm-modules/fmDNS/classes/class_buildconf.php
+++ b/server/fm-modules/fmDNS/classes/class_buildconf.php
@@ -929,6 +929,7 @@ class fm_module_buildconf extends fm_shared_module_buildconf {
                                                        $zones .= $this->getZoneOptions(array($zone_result[$i]->domain_id, $zone_result[$i]->parent_domain_id, $zone_result[$i]->domain_template_id), $server_serial_no, $domain_type, $server_group_ids). (string) $auto_zone_options;
                                                        /** Build zone file */
                                                        $zone_file_contents = ($domain_type == 'master') ? $this->buildZoneFile($zone_result[$i], $server_serial_no) : null;
+                                                 if($zone_file_contents!=null) //avoid truncation
                                                        $files[$zone_data_dir . '/' . $domain_type . '/' . $domain_name_file] = array('contents' => $zone_file_contents, 'syntax_check' => $zone_result[$i]->domain_check_config);
                                                        unset($zone_file_contents);
                                                        break;
```

**Comment by WillyXJ** [comment]
I think you're right and thanks for the patch.

**Comment by WillyXJ** [comment]
This is now fixed in [fmDNS 5.3.1](http://www.facilemanager.com/modules/fmdns/) and later.
